docs/source/models.rst

```diff
@@ -139,48 +139,134 @@ For more information, see the :mod:`mlflow.pyfunc` documentation.
+called. Therefore, the correct version of ``h2o(-py)`` must be installed in the loader's 
+environment. The arguments given to 
+`h2o.init() <http://docs.h2o.ai/h2o/latest-stable/h2o-py/docs/h2o.html#h2o.init>`_ can be 
+customized by modifying the ``init`` entry of the persisted H2O model's YAML configuration file: 
+``model.h2o/h2o.yaml``.
+
+Finally, the :py:func:`mlflow.h2o.load_model()` method can be used to load MLflow models with the
+``h2o`` flavor as H2O model objects.
+
+For more information, see :py:mod:`mlflow.h2o`.
 
 Keras (``keras``)
 ^^^^^^^^^^^^^^^^^
 
-The ``keras`` model flavor enables logging and loading Keras models. This model will be saved in a HDF5 file format, via the model_save functionality provided by Keras. Additionally, model can be loaded back as ``Python Function``. For more information, see :py:mod:`mlflow.keras`.
+The ``keras`` model flavor enables logging and loading Keras models. The :py:mod:`mlflow.keras`
+module defines :py:func:`save_model() <mlflow.keras.save_model>` and 
+:py:func:`log_model() <mlflow.keras.log_model>` functions that can be used to save Keras models
+in MLflow model format. These functions serialize Keras models as HDF5 files using the Keras
+library's built-in model persistence functions. MLflow models produced by these functions
+also contain the ``python_function`` flavor, allowing them to be interpreted as generic
+Python functions for inference via :py:func:`mlflow.pyfunc.load_pyfunc()`. Finally, the
+:py:func:`mlflow.keras.load_model` function can be used to load MLflow models with the
+``keras`` flavors as `Keras Model objects <https://keras.io/models/about-keras-models/>`_.
+
+For more information, see :py:mod:`mlflow.keras`.
 
 MLeap (``mleap``)
 ^^^^^^^^^^^^^^^^^
 
-The ``mleap`` model flavor supports saving models using the MLeap persistence mechanism. A companion module for loading MLflow models with the MLeap flavor format is available in the ``mlflow/java`` package. For more information, see :py:mod:`mlflow.mleap`.
+The ``mleap`` model flavor supports saving Spark models in MLflow format using the 
+`MLeap <http://mleap-docs.combust.ml/>`_ persistence mechanism. MLeap is an inference-optimized 
+format and execution engine for Spark models that does not depend on 
+`SparkContext <https://spark.apache.org/docs/latest/api/python/pyspark.html#pyspark.SparkContext>`_
+to evaluate inputs.
+
+Spark models can be saved with in MLflow format with MLeap by specifying the ``sample_input`` 
+argument of the :py:func:`mlflow.spark.save_model()` or :py:func:`mlflow.spark.log_model()` method 
+(recommended). The :py:mod:`mlflow.mleap` module also defines 
+:py:func:`save_model() <mlflow.mleap.save_model>` and 
+:py:func:`log_model() <mlflow.mleap.log_model>` methods for saving MLeap models in MLflow format, 
+but these methods do not include the ``python_function`` flavor in the models they produce.
+
+A companion module for loading MLflow models with the MLeap flavor format is available 
```

Comment:
A companion module for loading MLflow models with the MLeap flavor format  -> A companion module for loading MLflow model format  with the MLeap flavor

Reply:
Removed `format`. Thanks!